lib/jwt_sessions/store_adapters/abstract_store_adapter.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'redis'
+
+module JWTSessions
+  module StoreAdapters
+    class AbstractStoreAdapter
+      class << self
+        def instance_by_symbol(adapter, options = {})
+          camelized_adapter = adapter.to_s.split('_').map(&:capitalize).join
+          adapter_class_name = "JWTSessions::StoreAdapters::#{camelized_adapter}StoreAdapter"
+          Object.const_get(adapter_class_name).instance(options)
+        end
+
+        def instance(token_prefix: nil, **store_options)
+          @_tokens_store = configured_store(store_options)
+          @_token_prefix = token_prefix || JWTSessions.token_prefix
+
+          new(@_tokens_store, @_token_prefix)
+        end
+
+        private
+
+        def new(store, prefix)
+          super(store, prefix)
+        end
+      end
+
+      attr_reader :store, :prefix
+
+      def initialize(store, prefix)
+        @store  = store
+        @prefix = prefix
+      end
+
+      def fetch_access(uid)
+        csrf = store.get(access_key(uid))
+        return {} if csrf.nil?
+        { csrf: csrf }
+      end
+
+      def persist_access(uid, csrf, expiration)
+        key = access_key(uid)
+        store.set(key, csrf)
+        store.expireat(key, expiration)
+      end
+
+      def fetch_refresh(uid, namespace)
+        keys   = %i[csrf access_uid access_expiration expiration]
```

Comment:
Don't we want to move this into a constant and freeze it to avoid allocations?

Reply:
done